x-pack/plugins/security_solution/public/common/components/filter_group/index.tsx

```diff
@@ -128,9 +142,31 @@ const FilterGroupComponent = (props: PropsWithChildren<FilterGroupProps>) => {
     });
   }, [timeRange, filters, query, chainingSystem, controlGroup]);
 
+  const handleInputUpdates = useCallback(
+    (newInput: ControlGroupInput) => {
+      if (isEqual(controlGroupInputUpdates, newInput)) return;
+      if (
+        !isEqual(newInput.panels, controlGroupInputUpdates?.panels) &&
+        filterGroupViewMode === ViewMode.EDIT
```

Comment:
can this be made more readable?

Reply:
Yes, you are right. more changes incoming.. I should have made it draft ( apologies for that ) since the code is still in progress ( mainly housekeeping and Testing).

Will ping you again after these changes and thanks for your feedback.. will keep it in mind in further changes.